snap/snapcraft.yaml

```diff
@@ -0,0 +1,150 @@
+    restart-delay: 3s
+
+  otbr-nat44:
+    after: [otbr-setup]
+    command: bin/otbr-nat44 start
+    daemon: simple
+    install-mode: enable
+    restart-delay: 10s
+  
+parts:
+  openthread-deps:
+    source: https://github.com/openthread/ot-br-posix.git
+    source-tag: thread-reference-20230119
+    source-depth: 1
+    plugin: cmake
+    build-snaps:
+      - cmake
+    build-packages: 
+      - wget
+      - libreadline-dev
+      - libncurses-dev
+      - build-essential
+      - ninja-build
+      - libcpputest-dev
+      - libdbus-1-dev
+      - libavahi-common-dev 
+      - libavahi-client-dev
+      - libjsoncpp-dev
+      - libprotobuf-dev
+      - protobuf-compiler
+
+    stage-packages:
+      - iproute2
+      - iputils-ping
+      - rsyslog
+      - libavahi-client3
+      - iptables
+      - ipset
+      - dnsmasq-base
+      - dnsmasq-utils
+      - network-manager
+      - dhcpcd5
+      - libatm1
+
+    override-build: |
+      craftctl default
+      cd $CRAFT_PART_BUILD
+
+      # Install the mDNSResponder library on Linux system
+      MDNS_RESPONDER_SOURCE_NAME=mDNSResponder-1310.80.1
+      wget --no-check-certificate https://github.com/apple-oss-distributions/mDNSResponder/archive/refs/tags/$MDNS_RESPONDER_SOURCE_NAME.tar.gz
+      mkdir -p $MDNS_RESPONDER_SOURCE_NAME
+      tar xvf $MDNS_RESPONDER_SOURCE_NAME.tar.gz -C $MDNS_RESPONDER_SOURCE_NAME --strip-components=1
+      cd $CRAFT_PART_BUILD/$MDNS_RESPONDER_SOURCE_NAME/Clients
+      sed -i '/#include <ctype.h>/a #include <stdarg.h>' dns-sd.c
+      sed -i '/#include <ctype.h>/a #include <sys/param.h>' dns-sd.c
+      cd $CRAFT_PART_BUILD/$MDNS_RESPONDER_SOURCE_NAME/mDNSPosix
+      make os=linux
+      make install os=linux
+
+  ot-br-posix:
+    after: [openthread-deps]
+    plugin: nil
+    source: local/bin
+    build-packages: 
+      - sudo
+      - lsb-release
+      - kmod
+      - iptables
+      - xtables-addons-common
+    override-build: |
+      craftctl default
+      
+      cd $CRAFT_PART_BUILD
+
+      cp -vf ./otbr-build.sh ../../openthread-deps/src/script
+      cd ../../openthread-deps
+
+      # This script builds and installs border router and dependencies.
+      chmod +x ./src/script/otbr-build.sh
+      ./src/script/otbr-build.sh
+
+      # Todo: install -DT "$CRAFT_PART_SRC/LICENSE" "$CRAFT_PART_INSTALL/usr/share/doc/ot-br-posix/LICENSE"
+      # Copy library
+      mkdir -p $CRAFT_PART_INSTALL/usr/lib/
+      cp -v /usr/lib/libdns_sd.so $CRAFT_PART_INSTALL/usr/lib/
+
+      # Configure and copy executables
+      mkdir -p $CRAFT_PART_INSTALL/bin
+      cp ./build/otbr/src/agent/otbr-agent $CRAFT_PART_INSTALL/bin/
+      cp ./build/otbr/src/web/otbr-web $CRAFT_PART_INSTALL/bin/
+      cp ./build/otbr/third_party/openthread/repo/src/posix/ot-ctl $CRAFT_PART_INSTALL/bin/
+
+      chmod a+x ./script/otbr-firewall
+      cp ./src/script/otbr-firewall $CRAFT_PART_INSTALL/bin/
+
+      cp -v /etc/init.d/otbr-nat44 $CRAFT_PART_INSTALL/bin/
```

Comment:
After separating the `setup` script, the `otbr-nat44` script is missing. 
Need to check where the build process is.